impl/src/main/java/org/jboss/weld/resolution/QualifierInstance.java

```diff
@@ -134,7 +137,24 @@ public QualifierInstance(Class<? extends Annotation> annotationClass, Map<String
                     }
                     builder.put(method.getJavaMember().getName(), method.getJavaMember().invoke(instance));
                 } catch (IllegalAccessException | InvocationTargetException | IllegalArgumentException e) {
-                    throw ResolutionLogger.LOG.cannotCreateQualifierInstanceValues(instance, Formats.formatAsStackTraceElement(method.getJavaMember()), e);
+                    // it may happen that we are in EAR and have stored the annotation's method from different WAR class loader
```

Comment:
I think we should only re-try if `IllegalArgumentException` is thrown following the `java.lang.reflect.Method.invoke(Object, Object...)` javadoc.